msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -2291,5 +2394,33 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
         val result = application.signUp(invalidUsername)
         assertTrue(result is SignUpError)
         assertTrue((result as SignUpError).isInvalidUsername())
+
+        checkSafeLogging()
+    }
+
+    private fun checkSafeLogging() {
+        val piiTrueToCheck = listOf(
+            "password:","password=",
+            "code:","code=",
+            "continuationToken:","continuationToken="
+        )
+        val piiFalseToCheck = listOf(
+            "username:", "username="
+        )
+
+        val elementsToCheck = piiTrueToCheck.toMutableList()
+
+        if (!allowPII) {
+            elementsToCheck.addAll(piiFalseToCheck)
+        }
+
+        elementsToCheck.forEach { element ->
+            verifyLogDoesNotContain("Command", element)
+        }
+
+        Logger.getInstance().setExternalLogger(null)
+    }
+    private fun verifyLogDoesNotContain(tag: String, element: String) {
+        verify(externalLogger, never()).log(contains(tag), any(), contains(element), eq(true))
```

Comment:
I believe we want to do 2 types of checks:
- some values should *never* be in the callback
- some values may be, but only if also containsPII=true (and if the log configuration has allowPII enabled)

Reply:
Modify from eq(true) to eq(allowPII). However, the value of allowPII is defined variable of true status.